src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptionhandlers/ErrorNotificationExceptionHandler.java

```diff
@@ -1,38 +1,55 @@
 package uk.gov.hmcts.reform.bulkscanprocessor.exceptionhandlers;
 
 import com.microsoft.azure.servicebus.IMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import uk.gov.hmcts.reform.bulkscanprocessor.client.ErrorNotificationException;
-import uk.gov.hmcts.reform.bulkscanprocessor.model.ErrorNotificationMessageWrapper;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.servicebus.MessageAutoCompletor;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
 
 public class ErrorNotificationExceptionHandler {
 
-    public ErrorNotificationExceptionHandler() {
-        // empty construct
+    private static final Logger log = LoggerFactory.getLogger(ErrorNotificationExceptionHandler.class);
+
+    private final MessageAutoCompletor autoCompletor;
+
+    public ErrorNotificationExceptionHandler(MessageAutoCompletor autoCompletor) {
+        this.autoCompletor = autoCompletor;
     }
 
-    public ErrorNotificationMessageWrapper handle(IMessage message, Throwable throwable) {
+    public CompletableFuture<Void> handle(IMessage message, Throwable throwable) {
+        UUID lockToken = message.getLockToken();
         // return for message session completion step
         return throwable != null
-            ? handleNonNullThrowable(message, throwable)
-            : ErrorNotificationMessageWrapper.forAcknowledgement(message);
+            ? handleNonNullThrowable(lockToken, message.getMessageId(), throwable)
+            : autoCompletor.completeAsync(lockToken);
     }
 
-    private ErrorNotificationMessageWrapper handleNonNullThrowable(IMessage message, Throwable throwable) {
+    private CompletableFuture<Void> handleNonNullThrowable(UUID lockToken, String messageId, Throwable throwable) {
         if (throwable instanceof ErrorNotificationException) {
-            return handleErrorNotificationException(message, (ErrorNotificationException) throwable);
+            return handleErrorNotificationException(lockToken, messageId, (ErrorNotificationException) throwable);
         } else {
-            return ErrorNotificationMessageWrapper.forDeadLettering(message);
+            log.error("Unknown exception. Message ID: {}", messageId, throwable);
+
+            return autoCompletor.deadLetterAsync(lockToken, "Unknown exception", throwable.getMessage());
         }
     }
 
-    private ErrorNotificationMessageWrapper handleErrorNotificationException(
-        IMessage message,
+    private CompletableFuture<Void> handleErrorNotificationException(
+        UUID lockToken,
+        String messageId,
         ErrorNotificationException exception
     ) {
         if (exception.getStatus().is5xxServerError()) {
-            return ErrorNotificationMessageWrapper.forAbandoning(message);
+            log.warn("Received server error from notification client. Message ID: {}", messageId, exception);
```

Comment:
Is there a way to get the delivery count on from a queue message?
Could potentially be a useful information.

Reply:
I think there is a way. May be useful, but not crucial - we can still check how many times we failed processing.